GeneralsMD/Code/GameEngine/Source/Common/Thing/Thing.cpp

```diff
@@ -208,40 +208,26 @@ void Thing::setOrientation( Real angle )
 	Coord3D oldPos = m_cachedPos;
 	Matrix3D oldMtx = m_transform;
 
-	pos.x = m_transform.Get_X_Translation();
-	pos.y = m_transform.Get_Y_Translation();
-	pos.z = m_transform.Get_Z_Translation();
-	if( m_template->isKindOf( KINDOF_STICK_TO_TERRAIN_SLOPE) )
+	m_cachedAngle = normalizeAngle(angle);
+	Coord3D pos = { m_transform[0][3], m_transform[1][3] ,m_transform[2][3] };
+	m_cachedPos = pos;
+
+	if (m_template->isKindOf(KINDOF_STICK_TO_TERRAIN_SLOPE))
 	{
-		Matrix3D mtx;
-		const Bool stickToGround = true;	// yes, set the "z" pos				
```

Comment:
removing bool is just change for change sake, that said matrix removal is useful as it avoids unnecessary codegen